### ISSUE TITLE
Missing compile error

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#10231]
**TypeScript Version:**  2.0.0 beta & 1.8.10
**Code**

```ts
interface Type1<P1> {
}

interface Type2<P1, P2> {
}

interface Connection {
	sendMessage<P1>(type: Type1<P1>, p1: P1): void;
	sendMessage<P1, P2>(type: Type2<P1, P2>, p1: P1, p2: P2): void;
}

let connection: Connection;
let type: Type2<string, number>;

connection.sendMessage(type, 10);
```

**Expected behavior:**

connect.sendMessage is flagged as a compile error since the type is not matching. Looks like the compiler picks to use method `sendMessage<P1>(type: Type1<P1>, p1: P1): void;` however doesn't flag the mismatch between Type1 and Type2.

**Comment by RyanCavanaugh** [comment]
https://github.com/Microsoft/TypeScript/wiki/FAQ#why-is-astring-assignable-to-anumber-for-interface-at--

**Comment by dbaeumer** [comment]
@RyanCavanaugh thanks. Makes perfect sense.
